### PR TITLE
Applying documentation style to the Operator Guide

### LINKED ISSUE #17177: Conform Operator Guide to Keycloak style guidelines
### Description

Adjust wording and format of this guide to be consistent with other Keycloak guides.

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by andymunro**
Closes #17177
@ahus1, @stianst, @DGuhr, @pedroigor Please review

I had to change some Freemarker links to sections of the Server guide to be HTTPS links so they would work in RHBK.  However, they would still need to be converted to RHBK guidename links.  So I could revert those changes for now if you prefer.

### CODE REVIEW

**Review by stianst — CHANGES_REQUESTED** [review]
Please don't change the links. They are templated to use the correct links when publishing either to keycloak.org or to Pantheon.

File: docs/guides/src/main/operator/basic-deployment.adoc
Comment (ahus1) [inline docs/guides/src/main/operator/basic-deployment.adoc:26]:
```suggestion
The Keycloak Operator does not manage the database and you need to provision it yourself. Consider verifying your cloud provider offering or using a database operator such as https://access.crunchydata.com/documentation/postgres-operator/latest/[Crunchy].
```

File: docs/guides/src/main/operator/realm-import.adoc
Comment (ahus1) [inline docs/guides/src/main/operator/realm-import.adoc:17]:
```suggestion
* If a Realm with the same name already exists in Keycloak, it will not be overwritten.
```

File: docs/guides/src/main/operator/realm-import.adoc
Comment (ahus1) [inline docs/guides/src/main/operator/realm-import.adoc:19]:
```suggestion
* The Realm Import CR only supports creation of new realms and does not update or delete those. Changes to the realm performed directly on Keycloak are not synced back in the CR.
```

**Review by ahus1 — CHANGES_REQUESTED** [review]
Thank you Andy for reviewing these docs! I reviewed them from the contents perspective and they look good. 

+1 for Stian's comment about the links: Please keep the `<link...>` elements as is. I'll fix the links for the downstream docs by replacing them with appropriate downstream URL. 

See below for some minor comments. I also dared to suggest a comma: As I'm a non-native speaker, that's just a a suggestion and I leave the final decision to you.

**Comment by andymunro** [comment]
Thanks for the comments, including the comma, @ahus1, which I have addressed. I have also reverted the links to Freemarker format.

File: docs/guides/src/main/operator/basic-deployment.adoc
Comment (ahus1) [inline docs/guides/src/main/operator/basic-deployment.adoc:77]:
```suggestion
See <@links.server id="hostname"/> for the available configurations.
```

File: docs/guides/src/main/operator/customizing-keycloak.adoc
Comment (ahus1) [inline docs/guides/src/main/operator/customizing-keycloak.adoc:25]:
```suggestion
For instructions on how to build such an image, see <@links.server id="containers"/>.
```

File: docs/guides/src/main/operator/realm-import.adoc
Comment (ahus1) [inline docs/guides/src/main/operator/realm-import.adoc:41]:
```suggestion
The recommended way to obtain a `RealmRepresentation` is by leveraging the export functionality <@links.server id="importExport"/>.
```

**Review by ahus1 — CHANGES_REQUESTED** [review]
Hi @andymunro - thanks for reverting back the `<link...>` elements.

I noticed that one of them switch its ID. Please update it with the old ID. 

The other missing `.` where probably missing before anyway, and I might have missed them in the previous review.

Thanks!

**Comment by andymunro** [comment]
Thanks, @ahus1.  Corrections made plus a couple of other minor improvements.

**Review by ahus1 — DISMISSED** [review]
Thank you for those changes, they look good to me!

**Comment by ahus1** [comment]
@stianst - this is IMHO now ready to be merged. Thanks!

**Review by stianst — COMMENTED** [review]
LGTM, but @vmuzikar should probably take a look as well before we merge

File: docs/guides/src/main/operator/advanced-configuration.adoc
Comment (vmuzikar) [inline docs/guides/src/main/operator/advanced-configuration.adoc:68]:
```suggestion
* Fields that do not apply to a Kubernetes environment
```

Or maybe even something like:
```
Fields that are not relevant to a Kubernetes environment
```

File: docs/guides/src/main/operator/customizing-keycloak.adoc
Comment (vmuzikar) [inline docs/guides/src/main/operator/customizing-keycloak.adoc:18]: I would like to suggest the users what we consider a best practice. Without this heading, I thing the empasis on building a custom image is not as strong as it was.

**Review by vmuzikar — CHANGES_REQUESTED** [review]
@andymunro Added a few comments, thank you.

**Review by vmuzikar — APPROVED** [review]
LGTM, thanks @andymunro.

**Pull request merged**
